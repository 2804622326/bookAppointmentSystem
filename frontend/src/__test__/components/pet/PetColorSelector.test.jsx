import React from 'react';
import { render, waitFor, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import PetColorSelector from '../../../components/pet/PetColorSelector.jsx';

jest.mock('../../../components/pet/PetService.js', () => ({
  getAllPetColors: jest.fn(() => Promise.resolve({ data: ['Black'] }))
}));

test('loads colors and handles change', async () => {
  const onChange = jest.fn();
  const { asFragment } = render(<PetColorSelector value='' onChange={onChange} />);
  await waitFor(() => screen.getByText('Black'));
  fireEvent.change(screen.getByRole('combobox'), { target: { value: 'Black' } });
  expect(onChange).toHaveBeenCalled();
  expect(asFragment()).toMatchSnapshot();
});
