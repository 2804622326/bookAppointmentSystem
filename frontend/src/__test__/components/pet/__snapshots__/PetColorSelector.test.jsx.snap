// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`loads colors and handles change 1`] = `
<DocumentFragment>
  <div
    class="mb-2 col"
  >
    <select
      class="form-control"
      id="petColor"
      name="color"
      required=""
    >
      <option
        value=""
      >
        ...select pet color...
      </option>
      <option
        value="Black"
      >
        Black
      </option>
      <option
        value="add-new-item"
      >
        Add New
      </option>
    </select>
  </div>
</DocumentFragment>
`;
